src/lib/InstantSearch.ts

```diff
@@ -184,6 +185,18 @@ See: https://www.algolia.com/doc/guides/building-search-ui/going-further/backend
       );
     }
 
+    warning(
+      Boolean(insightsClient) || !hasDetectedInsightsClient(),
+      `InstantSearch detected the insights client in the global scope. 
+To connect InstantSearch to insights client, make sure you provide it to your InstantSearch instance:
+
+const search = instantsearch({
+  /* ... */
+  insightsClient: window.aa,
+});
+`
```

Comment:
We could link to the option in the [API reference](https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/). However I realize that we haven't documented `insightsClient` there.

Reply:
> We could link to the option in the API reference.

True, but I wasn't sure how to point to:
- `https://www.algolia.com/doc/api-reference/widgets/instantsearch/js` for vanilla
- `https://www.algolia.com/doc/api-reference/widgets/instantsearch/angular` for angular
- etc.

> I realize that we haven't documented insightsClient there.

Good point. Filed a task here: https://algolia.atlassian.net/browse/IFW-1104

Reply:
Linking to the right flavor documentation is a recurring problem that we already have. The same goes for widget vs. connector. We might have to introduce an error boundary system at the widgets' level at some point but this is beyond this scope. For consistency, we could link to the vanilla documentation once available (that's not a blocker for this PR anyway).

Reply:
done using `withUsage` here  b4b38a9